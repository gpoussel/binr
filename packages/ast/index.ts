export * from "./src/evaluation";
export * from "./src/nodes";
export * from "./src/visitor";
