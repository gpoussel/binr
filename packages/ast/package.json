{
  "name": "@binr/ast",
  "version": "1.0.0",
  "description": "AST classes for Binr application",
  "author": "Guillaume Poussel <gpoussel@gmail.com>",
  "homepage": "https://github.com/gpoussel/binr/tree/master/packages/ast#readme",
  "license": "MIT",
  "main": "index.js",
  "directories": {
    "src": "src"
  },
  "files": [
    "src"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gpoussel/binr.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1",
    "tsc": "tsc"
  },
  "bugs": {
    "url": "https://github.com/gpoussel/binr/issues"
  },
  "dependencies": {
    "@binr/shared": "^1.0.0"
  }
}
