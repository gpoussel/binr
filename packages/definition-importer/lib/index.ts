export * from "./common";
export * from "./formats";
export * from "./validation";
