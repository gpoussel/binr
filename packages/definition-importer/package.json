{
  "name": "@binr/definition-importer",
  "version": "1.0.0",
  "description": "Import definition from other formats",
  "author": "Guillaume Poussel <gpoussel@gmail.com>",
  "homepage": "https://github.com/gpoussel/binr/tree/master/packages/definition-importer#readme",
  "license": "MIT",
  "main": "lib/definition-importer.js",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "engines": {
    "node": ">=9.11.2"
  },
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/gpoussel/binr.git"
  },
  "scripts": {
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "bugs": {
    "url": "https://github.com/gpoussel/binr/issues"
  },
  "dependencies": {
    "chevrotain": "^4.7.0",
    "escape-string-regexp": "^2.0.0",
    "gunzip-maybe": "^1.4.1",
    "lerna": "^3.10.8",
    "lodash": "^4.17.11",
    "tar-stream": "^2.1.0"
  }
}
