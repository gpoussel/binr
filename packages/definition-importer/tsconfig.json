{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "./dist",
    "types": ["escape-string-regexp"]]
  },
  "include": ["./lib/**/*"]
}
