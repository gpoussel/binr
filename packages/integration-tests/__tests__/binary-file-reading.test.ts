import * as fs from "fs";

import { EvaluationContext } from "@binr/ast";
import { BinrDefinitionImporter } from "@binr/definition-importer";
import { each } from "lodash";

import { TestEvaluationInput } from "./test-evaluation-input";

const pathToDefinitionFixtures = `${__dirname}/../__fixtures__/definitions`;

const FILE_TYPES = [
  { type: "AVI", definitionFilename: "avi.binr", binaryFilename: "MVI_3572.avi" },
  { type: "EOT", definitionFilename: "eot.binr", binaryFilename: "roboto-regular-webfont.eot" },
  { type: "GIF", definitionFilename: "gif.binr", binaryFilename: "tree.gif" },
  { type: "ICO", definitionFilename: "ico.binr", binaryFilename: "bing.ico" },
  { type: "Java Class", definitionFilename: "java_class.binr", binaryFilename: "CucumberModules.class" },
  { type: "Android Manifest", definitionFilename: "manifest.binr", binaryFilename: "AndroidManifest.xml" },
  { type: "RAR", definitionFilename: "rar.binr", binaryFilename: "sample.rar" },
  { type: "RM video", definitionFilename: "rm.binr", binaryFilename: "video.rm" },
  { type: "ROMFS", definitionFilename: "romfs.binr", binaryFilename: "sample.romfs" },
  { type: "SHX", definitionFilename: "shx.binr", binaryFilename: "us_ski_areas.shx" },
  { type: "torrent", definitionFilename: "torrent.binr", binaryFilename: "ubuntu.torrent" },
  { type: "WMF", definitionFilename: "wmf.binr", binaryFilename: "sample.wmf" },
];

describe("Binary file reading", () => {
  const definitionReader = new BinrDefinitionImporter();

  each(FILE_TYPES, ({ type, definitionFilename, binaryFilename }) => {
    test(`reads ${type} file (${binaryFilename})`, () => {
      const definitionFile = fs.readFileSync(`${pathToDefinitionFixtures}/${definitionFilename}`);
      const definition = definitionReader.readInput(definitionFile.toString());

      const result = definition.evaluate(new EvaluationContext(), new TestEvaluationInput(binaryFilename));
      expect(result).toBeDefined();
      expect(result).toMatchSnapshot();
    });
  });
});
