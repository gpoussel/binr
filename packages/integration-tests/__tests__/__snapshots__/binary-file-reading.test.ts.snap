// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Binary file reading reads AVI file (MVI_3572.avi) 1`] = `Object {}`;

exports[`Binary file reading reads Android Manifest file (AndroidManifest.xml) 1`] = `Object {}`;

exports[`Binary file reading reads EOT file (roboto-regular-webfont.eot) 1`] = `Object {}`;

exports[`Binary file reading reads GIF file (tree.gif) 1`] = `Object {}`;

exports[`Binary file reading reads ICO file (bing.ico) 1`] = `Object {}`;

exports[`Binary file reading reads Java Class file (CucumberModules.class) 1`] = `Object {}`;

exports[`Binary file reading reads RAR file (sample.rar) 1`] = `Object {}`;

exports[`Binary file reading reads RM video file (video.rm) 1`] = `Object {}`;

exports[`Binary file reading reads ROMFS file (sample.romfs) 1`] = `Object {}`;

exports[`Binary file reading reads SHX file (us_ski_areas.shx) 1`] = `Object {}`;

exports[`Binary file reading reads WMF file (sample.wmf) 1`] = `Object {}`;

exports[`Binary file reading reads torrent file (ubuntu.torrent) 1`] = `Object {}`;
